dev_requirements.txt

```diff
@@ -0,0 +1,8 @@
+# Generated via tools/generate_requirements.py.
+# Do not edit this file; modify pyproject.toml instead and run the generator.
+mypy
```

Comment:
We can unpin this if tests pass, but if we get problems with it (Mypy is terrible backwards compat wise), we should put back a pin.

Reply:
Should the pin be added dynamically to `dev_requirements.txt`? I've restored the pin to the CI workflow.